### ISSUE TITLE
invalid times argument for packageBartResults with multiple chains and unmeasured levels

### ISSUE THREAD

**Issue by drcthal** [vdorie/dbarts#70]
When I run an rbart model with multiple chains and more than 2 levels in group.by.test that are not present in group.by, I am getting an error:
Error in rep(chainResults[[k]]$tau, rep_len(n.samples, n.unmeasured)) : invalid 'times' argument

This is happening inside dbarts:::packageBartResults on line 432 of dbarts/R/rbart.R, when the samples of tau are being repeated for each of the unmeasured levels to draw new random effects.
I believe simply changing from `rep(chainResults[[k]]$tau, rep_len(n.samples, n.unmeasured))` to `rep(chainResults[[k]]$tau, n.unmeasured)` would solve the issue.

To reproduce:
```
train <- data.frame(id=1:10, y = rnorm(10), x = rnorm(10))
test <- data.frame(id=11:13, x=rnorm(3))
dbarts::rbart_vi(y ~ x, data=train, test=test, group.by=train$id, group.by.test=test$id, n.samples=10, n.burn=10, n.chains=4, n.threads=1)
```

**Comment by vdorie** [comment]
Thanks! Hopefully fixed in eeb901d.

**Comment by drcthal** [comment]
Thanks for the incredibly quick fix! And for catching my goof with the lack of `each` in my proposed change.
